tests/unit/build/src/CMakeLists.txt

```diff
@@ -26,6 +25,8 @@ if(EXISTS "${HPX_DIR}")
   )
 
   add_executable(test_executable test.cpp)
+  target_link_libraries(test_executable ${HPX_LIBRARIES} iostreams_component test_component)
+  target_compile_definitions(test_executable PUBLIC -DHPX_DEBUG)
   hpx_setup_target(test_executable
     DEPENDENCIES test_component
     COMPONENT_DEPENDENCIES iostreams
```

Comment:
No, hpx_setup target is clever enough to automatically determine the type of the target. That is it can figure out if the target builds a executable or library, if it builds a library, the only thing left to do is specifying if it is just a plain library or a component library and if it is a plugin.